WordPress/Classes/ViewRelated/Comments/CommentsViewController.m

```diff
@@ -36,20 +37,22 @@ - (void)dealloc
     _tableViewHandler.delegate = nil;
 }
 
-- (instancetype)init
++ (CommentsViewController *)controllerWithBlog:(Blog *)blog
 {
-    self = [super init];
-    if (self) {
-        self.restorationClass = [self class];
-        self.restorationIdentifier = NSStringFromClass([self class]);
-    }
-    return self;
+    NSParameterAssert([blog isKindOfClass:[Blog class]]);
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"CommentsList" bundle:nil];
+    CommentsViewController *controller = [storyboard instantiateInitialViewController];
```

Comment:
Cool that we're setting the restoration identifier via the storyboard, but for state restoration to work we'd need to still set the restoration class. 
```
controller.restorationClass = [controller class];
```
However, when I tested this it resulted in a crash due to `self.blog` being nil after restoration.  The methods to encode the blog were apparently never implemented, and since there was no prior crash I'm doubtful state restoration was working correctly. 🤔 

To wire up restoration we'd need to implement something like: 
```
-(void)encodeRestorableStateWithCoder:(NSCoder *)coder
{
    [coder encodeObject:[self.blog hostname] forKey:@"blog-hostname"];
    [super encodeRestorableStateWithCoder:coder];
}

- (void)decodeRestorableStateWithCoder:(NSCoder *)coder
{
    NSString *hostname = [coder decodeObjectForKey:@"blog-hostname"];
    BlogService *service = [BlogService serviceWithMainContext];
    self.blog = [service blogByHostname:hostname];
    [super decodeRestorableStateWithCoder:coder];
}
```
However, this still had an issue as `[self viewDidAppear]` is executed prior to `[self decodeRestorableStateWithCoder]` and `self.blog` was still nil.  

Adding a check for the `blog` in `viewDidAppear` and implementing 
```
- (void)applicationFinishedRestoringState
{
    [super applicationFinishedRestoringState];
    [self refreshNoResultsView];
    [self refreshAndSyncIfNeeded];
}
```
Avoided the crash, however no results were being returned.  I'm assuming because `fetchRequest` captured the nil blog and needed to be reset but this is the point where I stopped tinkering. 

I think since restoration was probably busted already we can just make a note of all of this and address later (as alluded to in Slack). :)